File: svc-event/src/main/java/com/beancounter/event/contract/CorporateEventResponse.java
package com.beancounter.event.contract;

import com.beancounter.common.contracts.Payload;
import com.beancounter.common.event.CorporateEvent;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@Builder
@RequiredArgsConstructor
@AllArgsConstructor
public class CorporateEventResponse implements Payload<CorporateEvent> {
  private CorporateEvent data;
}

File: svc-event/src/main/kotlin/com/beancounter/event/contract/CorporateEventResponse.kt
package com.beancounter.event.contract

import com.beancounter.common.contracts.Payload
import com.beancounter.common.event.CorporateEvent

/**
 * Payload representation of a Corporate Action event.
 */
data class CorporateEventResponse(override val data: CorporateEvent) : Payload<CorporateEvent>
